tests/suite-unit/PluginFlyvemdmAgent.php

```diff
@@ -35,6 +35,22 @@
 
 class PluginFlyvemdmAgent extends CommonTestCase {
 
+   /**
+    * @var \PluginFlyvemdmAgent
+    */
+   private $enrolledAgent = null;
+
+   /**
+    * Enrol an Agent for tests which need data
+    * @return \PluginFlyvemdmAgent
+    */
+   private function enrolTestAgent() {
```

Comment:
This method creates a potential dependency between tests. You should avoid it. Just create a new agent anytime you call this method.

Moreover when the tests of this file runs in parallell you have one process per test method and you cannot share the instance of the agent.

Reply:
The idea of this method is for performance and only register one Agent on the DB for tests which need read a "real" data like showing a form or get mqtt topic. We change it to a mocked instance when DI is ready.

Reply:
> The idea of this method is for performance and only register one Agent on the DB

You don't have the expected optimization because the tests runs in parallel, in different processes. Then each test method (except one which has @engine inline annotation ) run in a distinct instance of the test class, each in a distinct process. 

please remove this method. Parallelization in atoum already speeds up things.